src/careamics/transforms/tta.py

```diff
@@ -1,74 +1,88 @@
 """Test-time augmentations."""
 
-from typing import List
-
 from torch import Tensor, flip, mean, rot90, stack
 
 
-# TODO add tests
 class ImageRestorationTTA:
     """
     Test-time augmentation for image restoration tasks.
 
     The augmentation is performed using all 90 deg rotations and their flipped version,
     as well as the original image flipped.
 
-    Tensors should be of shape SC(Z)YX
+    Tensors should be of shape SC(Z)YX.
 
     This transformation is used in the LightningModule in order to perform test-time
-    agumentation.
+    augmentation.
     """
 
-    def __init__(self) -> None:
-        """Constructor."""
-        pass
-
-    def forward(self, x: Tensor) -> List[Tensor]:
+    def forward(self, input_tensor: Tensor) -> list[Tensor]:
         """
         Apply test-time augmentation to the input tensor.
 
         Parameters
         ----------
-        x : Tensor
+        input_tensor : Tensor
             Input tensor, shape SC(Z)YX.
 
         Returns
         -------
-        List[Tensor]
+        list of torch.Tensor
             List of augmented tensors.
         """
+        # axes: only applies to YX axes
+        axes = (-2, -1)
+
         augmented = [
-            x,
-            rot90(x, 1, dims=(-2, -1)),
-            rot90(x, 2, dims=(-2, -1)),
-            rot90(x, 3, dims=(-2, -1)),
+            # original
+            input_tensor,
+            # rotations
+            rot90(input_tensor, 1, dims=axes),
+            rot90(input_tensor, 2, dims=axes),
+            rot90(input_tensor, 3, dims=axes),
+            # original flipped
+            flip(input_tensor, dims=(axes[0],)),
+            flip(input_tensor, dims=(axes[1],)),
```

Comment:
This doesn't actually cover every unique transformation.
There are 8 unique transformations (some combinations result in the same transformation e.g. flip X followed by flip Y is equivalent to rotate 180).
To cover every possible combination, do all the rotations and then the combination of each rotation with a flip.
```python
augmented = [
   # original
   input_tensor,
   # rotations
   rot90(input_tensor, 1, dims=axes),
   rot90(input_tensor, 2, dims=axes),
   rot90(input_tensor, 3, dims=axes),
   # rotations + flip
   flip(input_tensor, dims=(-1, )),
   flip(rot90(input_tensor, 1, dims=axes), dims=(-1, )),
   flip(rot90(input_tensor, 2, dims=axes), dims=(-1, )),
   flip(rot90(input_tensor, 3, dims=axes), dims=(-1, )),
]
```

Reply:
Next lines: https://github.com/CAREamics/careamics/blob/cfe269fcdc72d02e2f65490ad656262d5ab79692/src/careamics/transforms/tta.py#L49

I didn't want to recompute the rotations again (which might not be computationally faster...).

Reply:
My mistake! I guess I read it too quickly

> - **How**: Explicitly write the augmentations

I partly assumed this from your description in the PR.

Reply:
Ahah ouch, point taken. 😄